### PR TITLE
[DOC] Improve splitters docstrings

### LINKED ISSUE #3071: [BUG] `SlidingWindowSplitter` and  `ExpandingWindowSplitter` only returns a single item for test split
**Describe the bug**
When splitting a dataset based on `SlidingWindowSplitter` or `ExpandingWindowSplitter`, the test set for splits has only the last test index instead of an array of indices. The train set is as expected. Check below for a sample output of the issue.

**To Reproduce**

```python
import pandas as pd
from sktime.forecasting.model_selection import SlidingWindowSplitter, ExpandingWindowSplitter

df = pd.DataFrame(list(range(0, 50)))

sliding_splitter = SlidingWindowSplitter(fh=5, window_length=10, step_length=1)
s = list(sliding_splitter.split(df))
print(f"First split of SlidingWindowSplitter: {s[0]}")

expanding_splitter = ExpandingWindowSplitter(fh=5, initial_window=5, step_length=2)
ss = list(expanding_splitter.split(df))
print(f"First split of ExpandingWindowSplitter: {ss[0]}")
```
Above will result in 
```
>>> First split of SlidingWindowSplitter: (array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]), array([14]))
>>> First split of ExpandingWindowSplitter: (array([0, 1, 2, 3, 4]), array([9]))
```

**Expected behavior**
For `SlidingWindowSplitter` the first split based on give arguments must be 
`(array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]), array([10, 11, 12, 13, 14])`

For `ExpandingWindowSplitter` the first split based on give arguments must be 
`(array([0, 1, 2, 3, 4]), array([5, 6, 7, 8, 9]))`

**Additional context**

**Versions**
<details>

System:
    python: 3.8.13 (default, Mar 28 2022, 11:38:47)  [GCC 7.5.0]
executable: <removed>
   machine: Linux-5.10.60.1-microsoft-standard-WSL2-x86_64-with-glibc2.17

Python dependencies:
          pip: 22.1.2
   setuptools: 61.2.0
      sklearn: 1.1.1
       sktime: 0.12.1
  statsmodels: 0.13.2
        numpy: 1.22.4
        scipy: 1.8.1
       pandas: 1.4.3
   matplotlib: 3.5.2
       joblib: 1.1.0
        numba: 0.55.2
     pmdarima: 1.8.5
      tsfresh: 0.19.0
</details>

### PR DESCRIPTION
**Pull request by khrapovs**
#### Reference Issues/PRs

Closes #3071 

#### What does this implement/fix? Explain your changes.

Improve splitters' docstrings. In particular,
- clarify test window's relationship with forecasting horizon
- add code examples to all four splitters

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [X] I've added unit tests and made sure they pass locally.
- [X] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

### CODE REVIEW

File: sktime/forecasting/model_selection/_split.py [hunk 1]
```diff
@@ -673,8 +673,11 @@ class CutoffSplitter(BaseSplitter):
     which using the notation provided in :class:`BaseSplitter`,
     can be written as :math:`\{k_1,\ldots,k_n\}` for integer based indexing,
     or :math:`\{t(k_1),\ldots,t(k_n)\}` for datetime based indexing.
-    Training window's last point is equal to the cutoff,
-    while test window starts from the next observation in `y`.
+    Training window's last point is equal to the cutoff.
+    Test window is defined by forecasting horizons
+    relative to the end of the training window.
+    It will contain as many indices
+    as there are forecasting horizons provided to the `fh` argument.
```
Comment (fkiraly) [inline sktime/forecasting/model_selection/_split.py:680]: Hm, I would be more precise with this one:

... as there are forecasting horizon elements in the `fh` argument.
For a forecasating horizon :math:`\{h_1,\ldots, h_H\}`, the training window will
consist of the indices :math:`\{k_n+h_1,\ldots, k_n+h_H\}`.

File: sktime/forecasting/model_selection/_split.py [hunk 2]
```diff
@@ -1034,7 +1048,12 @@ class SlidingWindowSplitter(BaseWindowSplitter):
 
     Split time series repeatedly into a fixed-length training and test set.
 
-    For example for `window_length = 5`, `step_length = 1` and `fh = 3`
+    Test window is defined by forecasting horizons
+    relative to the end of the training window.
+    It will contain as many indices
+    as there are forecasting horizons provided to the `fh` argument.
+
+    For example for `window_length = 5`, `step_length = 1` and `fh = np.arange(1, 4)`
```
Comment (fkiraly) [inline sktime/forecasting/model_selection/_split.py:1056]: I would replace this by `fh = [1, 2, 3]` to make it clearer (or there may be a low level of confusion about where it starts or ends - 4? 3?)

File: sktime/forecasting/model_selection/_split.py [hunk 3]
```diff
@@ -673,8 +673,11 @@ class CutoffSplitter(BaseSplitter):
     which using the notation provided in :class:`BaseSplitter`,
     can be written as :math:`\{k_1,\ldots,k_n\}` for integer based indexing,
```
Comment (fkiraly) [inline sktime/forecasting/model_selection/_split.py:674]: minor note: due to our recent discussion where we agreed that duplicate time index elements should be possible, the indices should be tuples, not sets - correct?

File: sktime/forecasting/model_selection/_split.py (hunk 1 above)
Comment (khrapovs) [reply sktime/forecasting/model_selection/_split.py:681]: Thanks! Added this sentence to all splitters.

File: sktime/forecasting/model_selection/_split.py (hunk 2 above)
Comment (khrapovs) [reply sktime/forecasting/model_selection/_split.py]: Curious, `fh` is fine with a list of integers, but `cutoffs` is not :(
Changed as requested.

File: sktime/forecasting/model_selection/_split.py (hunk 3 above)
Comment (khrapovs) [reply sktime/forecasting/model_selection/_split.py:674]: This specific line refers to cutoffs. At least in my mind it does not make any sense to repeat cutoff values. That would be a very strange cross-validation exercise.

File: sktime/forecasting/model_selection/_split.py (hunk 1 above)
Comment (fkiraly) [reply sktime/forecasting/model_selection/_split.py:681]: is it a true sentence, for all the splitters?

File: sktime/forecasting/model_selection/_split.py (hunk 2 above)
Comment (fkiraly) [reply sktime/forecasting/model_selection/_split.py]: that is odd. should there be coercion?

File: sktime/forecasting/model_selection/_split.py (hunk 3 above)
Comment (fkiraly) [reply sktime/forecasting/model_selection/_split.py:674]: of course. The point is about whether, still, it is better to keep using tuples everywhere, for consistency.

File: sktime/forecasting/model_selection/_split.py (hunk 1 above)
Comment (khrapovs) [reply sktime/forecasting/model_selection/_split.py:681]: Yes

**Comment by fkiraly** [comment]
The doctests seem to fail on Windows - any ideas why?

File: sktime/forecasting/model_selection/_split.py (hunk 3 above)
Comment (khrapovs) [reply sktime/forecasting/model_selection/_split.py]: Now it's all tuples

File: sktime/forecasting/model_selection/_split.py (hunk 2 above)
Comment (khrapovs) [reply sktime/forecasting/model_selection/_split.py]: https://github.com/alan-turing-institute/sktime/issues/3082

**Comment by khrapovs** [comment]
> The doctests seem to fail on Windows - any ideas why?

Sorry, no idea. (Un)fortunately I can not reproduce as I do not have Windows.

**Comment by fkiraly** [comment]
The reason seems to be different printout on windows, e.g., 

```
2022-07-24T10:18:04.2841046Z Expected:
2022-07-24T10:18:04.2841416Z     [(array([3, 4, 5]), array([7, 9]))]
2022-07-24T10:18:04.2842142Z Got:
2022-07-24T10:18:04.2842537Z     [(array([3, 4, 5], dtype=int64), array([7, 9], dtype=int64))]
```

The solution would be to create a print that is consistent across OS, or use doctest wildcards or escapes.

**Comment by khrapovs** [comment]
> The solution would be to create a print that is consistent across OS, or use doctest wildcards or escapes.

Ok, I played with `print`, `repr`, `np.int64`. None of this helps. Of course, I could run the print in a loop and use something like `.astype`, but then the example will loose its simplicity. I have added `# doctest: +SKIP` just to skip the doctest. Any smarter ideas?

**Review by fkiraly — APPROVED** [review]
No smarter ideas, sorry. I suppose this is good as is then?
